### PR TITLE
List scholarships translations

### PR DESCRIPTION
**Pull request by edwingl10**
## Motivation and Context

ListScholarships & scholarshipList component translations.

## How Has This Been Tested?
local

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] User visible change (users will notice UI or functional changes)

## Previewing Changes

1. Go to https://s4us-pr-811.onrender.com/scholarships.
2. click icon to see translations

## Checklist:

- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: src/pages/ListScholarships.jsx [hunk 1]
```diff
@@ -110,10 +112,13 @@ function ListScholarships() {
         <title>Search Scholarships</title>
```
Comment (josueetcom) [inline src/pages/ListScholarships.jsx:112]: This wasn't translated

File: src/pages/ListScholarships.jsx [hunk 2]
```diff
@@ -110,10 +112,13 @@ function ListScholarships() {
...
       </Helmet>
       <Typography variant="h3" component="h1" align="center">
-        Scholarships
+        {t('listScholarships.title')}
       </Typography>
       <FilterBar queryParams={params} {...{ setQueryParam }} />
```
Comment (josueetcom) [inline src/pages/ListScholarships.jsx:117]: The filter bar is missing translations

File: src/components/ScholarshipList.jsx [hunk 3]
```diff
@@ -84,11 +86,10 @@ function ScholarshipList({ noResultsNode, listFn }) {
 
 ScholarshipList.propTypes = {
   listFn: PropTypes.func,
-  noResultsNode: PropTypes.node,
+  noResultsNode: PropTypes.node.isRequired,
```
Comment (josueetcom) [inline src/components/ScholarshipList.jsx:89]: You could make this optional, default to `undefined` and conditionally render the Typography element with the translation.

File: src/i18n/translations/english.json [hunk 4]
```diff
@@ -10,11 +10,18 @@
     "browse": "Browse Scholarships",
     "add": "Add a Scholarship",
     "login": "Login",
+    "load": "Load More",
```
Comment (josueetcom) [inline src/i18n/translations/english.json:13]:
```suggestion
    "loadMore": "Load More",
```

File: src/i18n/translations/english.json [hunk 5]
```diff
@@ -10,11 +10,18 @@
...
     "subscribe": "SUBSCRIBE FOR UPDATES"
   },
 
   "constructAlert": {
     "title": "Warning",
     "description": "🚧 Website Actively Under-Construction! 🚧"
+  },
+
+  "listScholarships": {
+    "title": "Scholarships",
```
Comment (josueetcom) [inline src/i18n/translations/english.json:23]: You could have a `general.scholarships` or `common.scholarships` key for this so we don't have to repeat this multiple times.

File: src/i18n/translations/english.json [hunk 6]
```diff
@@ -10,11 +10,18 @@
...
+    "end": "End of results",
+    "none": "No Scholarships found"
```
Comment (josueetcom) [inline src/i18n/translations/english.json:25]:
```suggestion
    "none": "No scholarships found"
```

Comment (josueetcom) [inline src/i18n/translations/english.json:25]: `endOfResults` and `noScholarshipsFound` keys are a little clearer.

File: src/pages/ListScholarships.jsx (hunk 2 above)
Comment (edwingl10) [reply src/pages/ListScholarships.jsx:117]: Should we also translate all the grade and major options?

File: src/components/ScholarshipList.test.jsx [hunk 7]
```diff
@@ -24,9 +31,15 @@ const fakeFirstPageOfManyResults = new Promise((resolve) => {
...
 
 test('renders no results', async () => {
-  renderWithTheme(<ScholarshipList listFn={() => fakeNoResults} />, {
-    wrapper: MemoryRouter,
-  });
+  renderWithTheme(
+    <ScholarshipList
+      listFn={() => fakeNoResults}
+      noResultsNode={'no scholarships found'}
+    />,
```
Comment (josueetcom) [inline src/components/ScholarshipList.test.jsx:38]: You can revert the rest of these changes now that the node is no longer needed. What you could do though is add a test case or two for the existence/presence of the property.

File: src/pages/ListScholarships.jsx (hunk 2 above)
Comment (josueetcom) [reply src/pages/ListScholarships.jsx:117]: Yeah that would be good. Though I guess a separate PR is fine.

**Review by josueetcom — DISMISSED** [review]

**Review by josueetcom — DISMISSED** [review]

**Review by josueetcom — APPROVED** [review]

**Pull request merged**